src/vsr/grid.zig

```diff
@@ -765,76 +865,57 @@ pub fn GridType(comptime Storage: type) type {
             // Remove the "root" read so that the address is no longer actively reading / locked.
             grid.read_queue.remove(read);
 
-            // A valid block filled by storage means the reads for the address can be resolved.
-            if (read_block_valid(cache_block.*, .{
+            const result = read_block_validate(block.*, .{
                 .address = read.address,
                 .checksum = read.checksum,
-                .block_type = read.block_type,
-            })) {
-                grid.read_block_resolve(read, cache_block.*);
+            });
+
+            if (result == .valid) {
+                grid.read_block_resolve(read, .{ .valid = block.* });
```

Comment:
This is `grid.read_block_resolve(read, result);`, right?

Would the control flow we simpler as

```zig
if (result != valid) {

}
grid.read_block_resolve(read, result);
```

so that there's only one call to resolve?